ixmp/core/scenario.py

```diff
@@ -4,17 +4,7 @@
 from numbers import Real
 from os import PathLike
 from pathlib import Path
-from typing import (
-    Any,
-    Callable,
-    Dict,
-    Iterable,
-    List,
-    MutableSequence,
-    Optional,
-    Sequence,
-    Union,
-)
+from typing import Any, Callable, Iterable, MutableSequence, Optional, Sequence, Union
```

Comment:
Thanks! I might as well make the change. 

Is this something that could be caught and/or fixed automatically with the right (“pro-grade”) mypy/ruff config? 

PS. I think one reason I'd shied away from this change is that it's harder to remember a `Callable` as one of `collections` (is it?) rather than something used for `typing`. But I guess this is now official and standard.

Reply:
Sounds good. I'm trying to reproduce the failure on 3.13 with numpy locally, though no luck so far.

Reply:
For some reason, the CI workflow happily installs pyam-iamc 1.3.1, which is almost two years old by now. I think this forces pint 0.17, which is incompatible with numpy 2.1.3.

Reply:
We had something similar before where 1.3.1 was the last release that didn't pin some package...but on pyam's `main`, I don't see any pins that should concern us at first glance. I'll note, though, that they don't support Python 3.13 yet. Their PyPi wheel is marked `py3-non-any`, though, which I thought meant "works with any Python3.x".

Reply:
...and pyam-iamc won't be able to support Python3.13 since it relies on ixmp4 and that doesn't support it yet. I'll get to work on moving the mypy PR on ixmp4 along, and try to add support for 3.13 next.

Reply:
> the CI workflow happily installs pyam-iamc 1.3.1, which is almost two years old by now. I think this forces pint 0.17, which is incompatible with numpy 2.1.3.

Good catch. I've noticed this happening locally, but didn't check for it happening on CI. I guess our solution (as we've done in the past) is to add an extra work-around step to upgrade packages that are forced to old versions by pyam.

Reply:
I just pushed another work-around commit (thanks for removing the older ones) that re-upgrades pint. My experience locally was that the tests pass with the older pyam-iamc version. Let's see if that also happens on GHA.